src/middlewared/middlewared/plugins/nvmet/global.py

```diff
@@ -0,0 +1,275 @@
+import pathlib
+
+import middlewared.sqlalchemy as sa
+from middlewared.api import api_method
+from middlewared.api.current import (NVMetGlobalAnaEnabledArgs,
+                                     NVMetGlobalAnaEnabledResult,
+                                     NVMetGlobalEntry,
+                                     NVMetGlobalRDMAEnabledArgs,
+                                     NVMetGlobalRDMAEnabledResult,
+                                     NVMetGlobalSessionsArgs,
+                                     NVMetGlobalSessionsResult,
+                                     NVMetGlobalUpdateArgs,
+                                     NVMetGlobalUpdateResult)
+from middlewared.plugins.rdma.constants import RDMAprotocols
+from middlewared.service import SystemServiceService, ValidationErrors, private
+from middlewared.utils import filter_list
+from .constants import NVMET_SERVICE_NAME
+from .kernel import clear_config, load_modules, nvmet_kernel_module_loaded, unload_module
+from .mixin import NVMetStandbyMixin
+from .utils import uuid_nqn
+
+NVMET_DEBUG_DIR = '/sys/kernel/debug/nvmet'
+
+
+class NVMetGlobalModel(sa.Model):
+    __tablename__ = 'services_nvmet_global'
+
+    id = sa.Column(sa.Integer(), primary_key=True)
+    nvmet_global_basenqn = sa.Column(sa.String(255), default=uuid_nqn())
```

Comment:
```suggestion
    nvmet_global_basenqn = sa.Column(sa.String(255), default=uuid_nqn())
```
will be more correct (although it changes nothing)

Reply:
I'm sorry, I don't see the delta between those two lines.

Reply:
Sorry, updated

Reply:
Oh, so it will execute a callable if necessary?  I didn't realize that.

Reply:
I've restored the original code for now.

Found that just providing the callable yielded an exception
```
  File "/usr/lib/python3/dist-packages/middlewared/service/config_service.py", line 110, in config
    return await self._get_or_insert(self._config.datastore, options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/service/config_service.py", line 129, in _get_or_insert
    await self.middleware.call('datastore.insert', datastore, {})
  File "/usr/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/plugins/datastore/connection.py", line 106, in execute_write
    result = self.connection.execute(sql, binds)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 1365, in execute
    return self._exec_driver_sql(
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 1669, in _exec_driver_sql
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 1943, in _execute_context
    self._handle_dbapi_exception(
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 2124, in _handle_dbapi_exception
    util.raise_(
  File "/usr/lib/python3/dist-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "/usr/lib/python3/dist-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (sqlite3.ProgrammingError) Error binding parameter 1: type 'function' is not supported
[SQL: INSERT INTO services_nvmet_global (nvmet_global_basenqn, nvmet_global_kernel, nvmet_global_ana, nvmet_global_rdma, nvmet_global_xport_referral) VALUES (?, ?, ?, ?, ?)]
[parameters: (<function uuid_nqn at 0x7fd179c63e20>, 1, 0, 0, 1)]
(Background on this error at: https://sqlalche.me/e/14/f405)


```

Reply:
I fixed this in https://github.com/truenas/middleware/pull/16340, could you please rebase after I marge and try again?

Reply:
Absolutely.  Thanks!  (Awaiting merge...)